packages/orchestration/src/exos/chain-hub.js

```diff
@@ -591,27 +593,31 @@ export const makeChainHub = (zone, agoricNames, vowTools) => {
       return undefined;
     },
     /**
-     * @param {string} partialId CAIP-10 account ID or a Cosmos bech32 address
+     * @param {AccountIdArg | string} account CAIP-10 account ID or a
+     *   CosmosChainAddress or bech32
      * @returns {AccountId}
      * @throws {Error} if chain info not found for bech32Prefix
      */
-    resolveAccountId(partialId) {
-      const parsed = parseAccountId(partialId);
+    resolveAccountId(account) {
+      if (typeof account !== 'string') {
+        return `cosmos:${account.chainId}:${account.value}`;
+      }
+
+      const parsed = parseAccountIdArg(account);
       if ('namespace' in parsed) {
         // It is already fully qualified
-        return /** @type {AccountId} */ (partialId);
+        return /** @type {AccountId} */ (account);
       }
-
-      const reference = resolveCosmosChainId(partialId);
-      return `cosmos:${reference}:${partialId}`;
+      const reference = resolveCosmosChainId(account);
+      return `cosmos:${reference}:${account}`;
     },
     /**
      * @param {string} partialId CAIP-10 account ID or a Cosmos bech32 address
      * @returns {CosmosChainAddress}
      * @throws {Error} if chain info not found for bech32Prefix
      */
     makeChainAddress(partialId) {
-      const parsed = parseAccountId(partialId);
+      const parsed = parseAccountIdArg(partialId);
```

Comment:
if `makeChainAddress` takes only a string then this change isn't necessary.

did you want `makeChainAddress` to support `CosmosChainAddress`? (i.e. @param {AccountIdArg}`)

Reply:
> did you want makeChainAddress to support CosmosChainAddress? (i.e. @param {AccountIdArg}`)

Yes, I think that would be an improvement.